### PR TITLE
Ensure empty results don't error

### PR DESCRIPTION
**Pull request by jrbourbeau**
Looking through the code, I think this test will fail. This PR is to make sure queries that return an empty table result in an empty DataFrame, not an error being raised

### CODE REVIEW

**Comment by jrbourbeau** [comment]
@phobson would you mind taking a look at this when you get a chance?

File: dask_snowflake/core.py [hunk 1]
```diff
@@ -264,47 +267,40 @@ def read_snowflake(
...
-        # Create Blockwise layer
-        layer = DataFrameIOLayer(
-            output_name,
-            meta.columns,
-            batches_partitioned,
-            # TODO: Implement wrapper to only convert columns requested
-            partial(_fetch_batches, arrow_options=arrow_options),
-            label=label,
+    batch_types = set(type(b) for b in batches)
+    if len(batch_types) > 1 or next(iter(batch_types)) is not ArrowResultBatch:
```
Comment (phobson) [inline dask_snowflake/core.py:278]: Filtering the batches earlier is paying off already

File: dask_snowflake/tests/test_core.py [hunk 2]
```diff
@@ -64,6 +64,21 @@ def test_write_read_roundtrip(table, connection_kwargs, client):
...
+
+    result = read_snowflake(
+        f"SELECT * FROM {table} where A > %(target)s",
+        execute_params={"target": df.A.max()},
+        connection_kwargs=connection_kwargs,
+        npartitions=2,
+    )
+    assert type(result) is dd.DataFrame
+    assert len(result.index) == 0
+    assert len(result.columns) == 0
```
Comment (phobson) [inline dask_snowflake/tests/test_core.py:79]: It'd be nice if we could match the columns in the empty dataframe, but I'll file that in the I-also-want-a-pony feature list.
image

**Review by phobson — APPROVED** [review]
Looks great. Nice thinking ahead on this.

**Comment by phobson** [comment]
@jrbourbeau, I'll leave it to you to smash the green button

**Comment by jrbourbeau** [comment]
Thanks for reviewing @phobson!

**Pull request merged**